scripts/viz/param_sweep_latent_space_viz.R

```diff
@@ -0,0 +1,321 @@
+#               --results_directory "z_param_sweep_tybalt"
+#               --output_filename "z_parameter_sweep_tybalt_full_results.tsv"
+#
+# Also perform the same for the ADAGE sweep
+#
+# Output:
+# Several figures describing the results of a parameter sweep over latent space
+# dimensionality for both Tybalt and ADAGE models
+
+library(dplyr)
+library(ggplot2)
+
+set.seed(123)
+
+# Set input and output file names
+tyb_file <- file.path("results", "z_parameter_sweep_tybalt_full_results.tsv")
+adage_file <- file.path("results", "z_parameter_sweep_adage_full_results.tsv")
+
+tyb_fig <- file.path("figures", "param_sweep", "z_param_tybalt")
+adage_fig <- file.path("figures", "param_sweep", "z_param_adage")
+
+# Read and process the tybalt latent space parameter sweep
+tybalt_df <- readr::read_tsv(tyb_file,
+                             col_types = readr::cols(
+                               .default = readr::col_character(),
+                               train_epoch = readr::col_integer(),
+                               loss = readr::col_double(),
+                               val_loss = readr::col_double()))
+
+# Melt the data frame with loss and val loss as measurement variables
+tybalt_melt_df <- reshape2::melt(tybalt_df,
+                                 id.vars = c("learning_rate", "batch_size",
+                                             "epochs", "kappa", "train_epoch",
+                                             "num_components"),
+                                 measure.vars = c("loss", "val_loss"),
+                                 variable.name = "loss_type",
+                                 value.name = "loss")
+
+# Get the validation loss at epoch end for each hyperparameter combination
+tybalt_select_df <- tybalt_melt_df %>%
+  dplyr::filter(loss_type == "val_loss") %>%
+  dplyr::group_by(learning_rate, batch_size, epochs, kappa, num_components) %>%
+  dplyr::summarize(end_loss = tail(loss, 1))
+
+# Order batch size and epoch variables
+tybalt_select_df$batch_size <- factor(tybalt_select_df$batch_size,
+                                      levels = c(50, 100, 150))
+tybalt_select_df$epochs <- factor(tybalt_select_df$epochs ,
+                                  levels = c(50, 100))
+
+# Recode batch size and learning rate variables for plotting
+tybalt_select_df$batch_size <-
+  dplyr::recode_factor(tybalt_select_df$batch_size, 
+                       `50` = "Batch: 50",
+                       `100` = "Batch: 100",
+                       `150` = "Batch: 150")
+
+tybalt_select_df$learning_rate <-
+  dplyr::recode_factor(tybalt_select_df$learning_rate, 
+                       `0.0005` = "Learning: 0.0005",
+                       `0.001` = "Learning: 0.001", 
+                       `0.0015` = "Learning: 0.0015",
+                       `0.002` = "Learning: 0.002",
+                       `0.0025` = "Learning: 0.0025")
+
+base_theme <-  theme(axis.text = element_text(size = rel(0.5)),
```

Comment:
In my opinion, I think this would be better placed before you read in files. Almost like it's a function. I found myself looking around for `base_theme` once I got to later plots.

Reply:
Yeah, I agree. Updated in next commit!